onnxruntime/core/codegen/common/utils.h

```diff
@@ -19,4 +20,12 @@ int64_t TotalSize(const std::vector<int64_t>& shape);
 
 void GetStrides(const int64_t* shape, int ndim, std::vector<int64_t>& strides);
 
+struct TargetFeature {
+  bool hasAVX;
+  bool hasAVX2;
+  bool hasAVX512;
```

Comment:
I tend to think we need to add a constructor which sets all class members to false. It would reduce the change of making mistakes, e.g. we create a TargetFeature object which has random true/false values for its members.